### ISSUE TITLE
[a11y] Add icon for external links

### ISSUE THREAD

**Issue by sarahill** [Shopify/polaris-react#944]
## TL;DR

Currently, there are a lot of different icons being used to signify an external link. These should be replaced with the new standard inline icon.

Based on new guideline and icons developed here: Shopify/polaris-icons#40

Buttons and links have an `external` prop. Add the updated external link icons when `external={true}`

![image](https://user-images.githubusercontent.com/8629173/42649988-30f92314-85d9-11e8-818a-80fc93b2e3e3.png)

Related issues: Shopify/polaris-ux/issues/78 and Shopify/polaris-icons/issues/40

**Comment by dpersing** [comment]
This would also help address https://github.com/Shopify/polaris-ux/issues/146, if we include a text equivalent along with the icon.

**Comment by ry5n** [comment]
I believe this was fixed in https://github.com/Shopify/polaris-react/pull/1284
